src/Factories/FunFunFactory.php

```diff
@@ -1428,6 +1434,13 @@ public function getCountries(): string {
 		) )->readAndValidateJson();
 	}
 
+	public function getValidationRules(): string {
+		return ( new JsonStringReader(
+			$this->getI18nDirectory() . '/data/validation.json',
+			new SimpleFileFetcher()
+		) )->readAndValidateJson();
+	}
+
```

Comment:
The JsonStringReader is on its way out (see https://phabricator.wikimedia.org/T254880) and its primary function is to provide *unparsed* JSON strings (instead of data structured) to the templating engine. Please change move the `json_decode` call to this method. You can still use `JsonStringReader` to make sure the JSON is valid, but you can also assume it is.